### ISSUE TITLE
Filters are applied inconsistently in web and Android app

### ISSUE THREAD

**Issue by bocops** [mastodon/mastodon#31156]
### Steps to reproduce the problem

1. In the web interface, create a filter for the phrase "mastodon" in public timelines, with "exact match" set to off.
2. In the web interface, navigate to the account `@NDR@ard.social` and observe that all its recent posts are visible.
3. Follow the account.
4. In the Android app, navigate to the same account and observe that all posts are hidden.
5. Wait for a new post from the account, observe that it is visible in the home timeline in the web interface, but hidden in the Android app.

### Expected behaviour

Filter should be applied the same across all interfaces

### Actual behaviour

Posts are hidden in some but not all interfaces

### Detailed description

For what it is worth, none of the filtered posts of the account seem to contain the phrase "mastodon", so it seems to be the handling in the Android app that seems to be at fault. This is hard to tell exactly, though.

Account given and exact filter phrase is just one example, other combinations lead to the same result.

### Mastodon instance

multiple; confirmed on at least det.social and nrw.social

### Mastodon version

v4.2.10

### Browser name and version

various Chromium-based

### Operating system

multiple/does not apply

**Comment by bocops** [comment]
After checking some of the filtered posts in question, it seems that they are filtered because they typically contain a link with a query parameter for tracking purposes. A recent example link is `https://www.ndr.de/ratgeber/gesundheit/Badeunfaelle-Wie-schuetzt-man-sich-vor-dem-Ertrinken-,baderegeln114.html?at_medium=mastodon&at_campaign=NDR.de`.

I think it is at least worth discussing whether an otherwise invisible query parameter should even lead to a post being filtered - but that's not the point of this issue.

The point here is that retrieving the post in question (`https://mastodon.social/@NDR@ard.social/112886825920794738`) via API while signed in like so: `https://mastodon.social/api/v1/statuses/112886825920794738` leads to a response containing the `filtered` attribute pasted below. It probably should be filtered in the web interface, but isn't.

```
"filtered": [
    {
      "filter": {
        "id": "***",
        "title": "mastodon test",
        "context": [
          "public"
        ],
        "expires_at": null,
        "filter_action": "warn"
      },
      "keyword_matches": [
        "mastodon"
      ],
      "status_matches": null
    }
  ],
```

**Comment by ThisIsMissEm** [comment]
Yep, keyword filters currently match in URLs too. Similar to the issue in #27990
